src/test/java/edu/byu/cs/server/ServerTest.java

```diff
@@ -0,0 +1,153 @@
+                Arguments.of( "GET", "/api/admin/analytics/commit", "commitAnalyticsGet", ":option"),
+                Arguments.of("POST", "/api/admin/repo", "setRepoUrlAdmin", ":netid"),
+                Arguments.of( "GET", "/api/admin/honorChecker/zip", "honorCheckerZipGet", ":section"),
+                Arguments.of( "GET", "/api/admin/submissions/latest", "latestSubmissionsGet", ":count"),
+                Arguments.of( "GET", "/api/admin/submissions/student", "studentSubmissionsGet", ":netid")
+        );
+    }
+
+    public static Stream<Arguments> getEndpoints() {
+        return Stream.of(
+                Arguments.of( "GET", "/auth/callback", "callbackGet"),
+                Arguments.of( "GET", "/auth/login", "loginGet"),
+                Arguments.of("POST", "/auth/logout", "logoutPost"),
+
+                Arguments.of( "GET", "/api/config", "getConfigStudent"),
+                Arguments.of( "GET", "/api/latest", "latestSubmissionForMeGet"),
+                Arguments.of( "GET", "/api/me", "meGet"),
+                Arguments.of("POST", "/api/repo", "setRepoUrl"),
+                Arguments.of( "GET", "/api/submission", "submissionXGet"),
+                Arguments.of( "GET", "/api/submit", "submitGet"),
+                Arguments.of("POST", "/api/submit", "submitPost"),
+
+                Arguments.of( "GET", "/api/admin/config", "getConfigAdmin"),
+                Arguments.of("POST", "/api/admin/config/banner", "updateBannerMessage"),
+                Arguments.of("POST", "/api/admin/config/courseIds", "updateCourseIdsPost"),
+                Arguments.of( "GET", "/api/admin/config/courseIds", "updateCourseIdsUsingCanvasGet"),
+                Arguments.of("POST", "/api/admin/config/phases", "updateLivePhases"),
+                Arguments.of("POST", "/api/admin/config/phases/shutdown", "scheduleShutdown"),
+
+                Arguments.of( "GET", "/api/admin/submissions/active", "submissionsActiveGet"),
+                Arguments.of("POST", "/api/admin/submissions/approve", "approveSubmissionPost"),
+                Arguments.of( "GET", "/api/admin/submissions/latest", "latestSubmissionsGet"),
+                Arguments.of("POST", "/api/admin/submissions/rerun", "submissionsReRunPost"),
+                Arguments.of("POST", "/api/admin/submit", "adminRepoSubmitPost"),
+
+                Arguments.of( "GET", "/api/admin/analytics/commit", "commitAnalyticsGet"),
+                Arguments.of( "GET", "/api/admin/repo/history", "repoHistoryAdminGet"),
+                Arguments.of( "GET", "/api/admin/sections", "sectionsGet"),
+                Arguments.of( "GET", "/api/admin/test_mode", "testModeGet"),
+                Arguments.of( "GET", "/api/admin/users", "usersGet")
+                );
+    }
+
+    @AfterAll
+    static void stopServer() {
+        server.stop();
+    }
+
+    @BeforeAll
+    public static void init() {
+        server = new Server(mockedMockProvider);
+        int port = server.start(8080);
```

Comment:
Could this be port 0 instead of hardcoding 8080? (0 lets the OS choose any open port)

Reply:
Done. I overloaded Server.start() so it's only in one place as well.